src/extension.ts

```diff
@@ -26,8 +53,113 @@ export function activate(context: vscode.ExtensionContext) {
     )
   );
 
+  // Automatically open Live Interview + Sidebar
   vscode.commands.executeCommand("hatchways-live-interviewing.welcome");
   vscode.commands.executeCommand("hatchways-sidebar.focus");
+
+  // When user submitted their name
+  socket.on(USER_READY, (message) => {
+    vscode.window.showInformationMessage(message);
+  });
+
+  // When user left
+  socket.on(DISCONNECT, (message) => {
+    vscode.window.showInformationMessage(message);
+  });
+
+  // When user open a file
+  let disposableCurrFileDecorationProviders: any = [];
+  socket.on(FILE_CLICK, (value) => {
+    for (const d of disposableCurrFileDecorationProviders) {
+      d.dispose();
+    }
+
+    // For each online user, register a FileDecorationProvider
+    disposableCurrFileDecorationProviders = [];
```

Comment:
What I discover is that `FileDecorationProvider` actually loops through all the existing files and check whether a decoration needed to be add. And a decoration can only be a single letter, say "A", or "B" so I have to loop through each online users here. 

There's a bug where if a user hasn't opened a file yet, it won't show the decoration on it. Not sure what's the best approach here 🤔 Unless we actually are okay with just automatically open a file for userB if userA click on it.

Reply:
>  automatically open a file for userB if userA click on it.

I think we don't want to do that :thinking:

Reply:
What specifically changes after someone has opened a file once vs not? Is there a way we could pre-initialize all the files as opened or something? :sweat_smile: